src/shared/metadata/cgroup_path_resolver.cc

```diff
@@ -26,20 +27,30 @@
 #include "src/common/fs/fs_wrapper.h"
 #include "src/shared/metadata/cgroup_path_resolver.h"
 
+#define S_MAGIC_CGROUP2FS 63677270
+DEFINE_bool(test_cgroup2_path, false, "Flag to force assume cgroup2 fs for testing purposes");
+
 namespace px {
 namespace md {
 
-StatusOr<std::string> CGroupBasePath(std::string_view sysfs_path) {
+StatusOr<std::string> CGroupBasePath(std::string_view sysfs_path, bool test_cgroup2_path) {
   // Different hosts may mount different cgroup dirs. Try a couple for robustness.
   const std::vector<std::string> cgroup_dirs = {"cpu,cpuacct", "cpu", "pids"};
 
   for (const auto& cgroup_dir : cgroup_dirs) {
     std::filesystem::path dir;
+    struct statfs info;
 
     // Attempt assuming naming scheme #1.
     std::string base_path = absl::StrCat(sysfs_path, "/cgroup/", cgroup_dir);
+    std::string cgv2_base_path = absl::StrCat(sysfs_path, "/cgroup");
+
     if (fs::Exists(base_path)) {
       return base_path;
+    } else if (((statfs(cgv2_base_path.c_str(), &info) == 0) &&
+                (info.f_type == S_MAGIC_CGROUP2FS)) ||
+               test_cgroup2_path) {
```

Comment:
Also:
1. we should make this the `if` clause, because it is the more specific condition we are testing for.
2. For readability, we should move the `stat` call out of the if statement and store the result to a variable. I think if we test for the filesystem of `absl::StrCat(sysfs_path, "/cgroup")` it should work for correctly identifying both v1 and v2.
3. As a follow-up to (2), we should even move the `stat` call out of the loop. We only need to do that once.

Reply:
Thanks @oazizi000 for the comments. I have a question if we do not pipe this as a boolean argument and instead reference the Flag directly wouldnt that render the `FLAG_force_cgroup2_mode` true for every test other test and we'll be returning `cgv2_base_path` for all the tests in the test suite ?

Can you please elaborate how this flag will come in action in tests for cgroup2 and not every other test ?

Reply:
Changes suggested for stat call fixed in https://github.com/pixie-io/pixie/pull/635/commits/b976406be7fed6f63bebd138f6ba4a0a1021abf6